keps/sig-network/4974-deprecate-endpoints/README.md

```diff
@@ -0,0 +1,612 @@
+# KEP-4974: Deprecate v1.Endpoints and Associated Controllers
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Formal Deprecation of <code>v1.Endpoints</code>](#formal-deprecation-of-v1endpoints)
+  - [Warnings via metrics](#warnings-via-metrics)
+  - [Documentation Updates](#documentation-updates)
+  - [<code>kubernetes.default</code> Endpoints](#kubernetesdefault-endpoints)
+  - [Endpoints Cleanup](#endpoints-cleanup)
+  - [Update Remaining Internal Endpoints Users](#update-remaining-internal-endpoints-users)
+  - [E2E Test Updates](#e2e-test-updates)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+      - [Unit tests](#unit-tests)
+      - [Integration tests](#integration-tests)
+      - [e2e tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Deprecation - Stage 1](#deprecation---stage-1)
+    - [Deprecation - Stage 2](#deprecation---stage-2)
+    - [Deprecation - Stage 3](#deprecation---stage-3)
+    - [Deprecation - Stage 4](#deprecation---stage-4)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
```

Comment:
nit: you should mark the required ones off that you can.

Reply:
We really need to move this checklist out of the KEP body... For example, the first checkbox ought to be checked immediately after the KEP is first merged (with that box not checked)...